conans/model/build_info.py

```diff
@@ -35,6 +35,56 @@ def __copy__(self):
         return the_copy
 
 
+class BuildModulesDict(dict):
+    """
+    A dictionary with append and extend for cmake build modules to keep it backwards compatible
+    with the list interface
+    """
+
+    def __getitem__(self, key):
+        if key not in self.keys():
+            super(BuildModulesDict, self).__setitem__(key, list())
+        return super(BuildModulesDict, self).__getitem__(key)
+
+    def append(self, item):
+        if item.endswith(".cmake"):
+            self["cmake"].append(item)
+            self["cmake_multi"].append(item)
+            self["cmake_find_package"].append(item)
+            self["cmake_find_package_multi"].append(item)
+
+    def extend(self, items):
+        for item in items:
+            self.append(item)
+
+    @classmethod
+    def from_list(cls, build_modules):
+        the_dict = BuildModulesDict({"cmake": [], "cmake_multi": [], "cmake_find_package": [],
+                                     "cmake_find_package_multi": []})
+        for item in build_modules:
+            if item.endswith(".cmake"):
+                for val in the_dict.values():
+                    val.append(item)
+        return the_dict
+
+    def to_abs_paths(self, rootpath):
+        for generator, values in self.items():
+            self[generator] = [os.path.join(rootpath, p) if not os.path.isabs(p) else p
+                               for p in values]
```

Comment:
If this `BuildModulesDict` should be a simple dictionary (your comment below), better not to add methods to this class that will be removed. We can just make it a free function.